prov/psm/src/psmx_fabric.c

```diff
@@ -103,10 +100,23 @@ int psmx_fabric(struct fi_fabric_attr *attr,
 	if (!fabric_priv)
 		return -FI_ENOMEM;
 
+	psmx_get_uuid(fabric_priv->uuid);
+	if (psmx_env.name_server) {
+		fabric_priv->util_fabric.name_server =
+			calloc(1, sizeof(*fabric_priv->util_fabric.name_server));
+		if (!fabric_priv->util_fabric.name_server) {
+			free(fabric_priv);
+			return -FI_ENOMEM;
+		}
+		fabric_priv->util_fabric.name_server->ns_port =
+			psmx_uuid_to_port(fabric_priv->uuid);
+	}
```

Comment:
I thought to pass `ns_port` through `ofi_fabric_init` and if `ns_port` is specified, run name_server functionality, but I dropped that idea (it impacts all provider that uses utility fabric and don't want to use Name Server).
I'd like to propose the two approaches:
1. What about to define utility env variable like FI_PSM2_NAME_SERVER that manages name server functionality? If FI_UTIL_NAME_SERVER is defined, it means that utility should enable Name Server functionality. Btw, it involves that we should create utility provider (like psm, verbs and etc - it may help to add tracing of utility code) to be compliant with existing libfabric's env functionality.
2. Separate utility fabric and utility NS into two unrelated functionality. - This is more simple and make sense for me.

Please, provide your feedback which approach is more preferable for you or no one of existing is OK for you.